File: examples/stats/multi_ess.r
# Compute multivariate ESS using multiESS function of mcmcse package

## Load libraries

library(mcmcse)

## Read chains

chains <- read.table(file="chain01.csv", header=FALSE, sep=",")

## Compute multivariate ESS using INSE MC covariance estimation

print(multiESS(chains, covmat=mcse.initseq(chains)$cov))

## Compute multivariate ESS batch mean MC covariance estimation

print(multiESS(chains))

# print(multiESS(chains, covmat=mcse.multi(chains)$cov)) # Same as print(multiESS(chains))

File: examples/stats/multi_ess.py
# Compute multivariate ESS using multi_ess function based on eeyore

# %% Load packages

import numpy as np
import torch

from eeyore.stats import multi_ess

# %% Read chains

chains = torch.as_tensor(np.genfromtxt('chain01.csv', delimiter=','))

# %% Compute multivariate ESS using INSE MC covariance estimation

ess_val = multi_ess(chains)

print('Multivariate ESS using INSE MC covariance estimation: {}'.format(ess_val))
